lnrpc/routerrpc/router.proto

```diff
@@ -60,24 +60,54 @@ message PaymentRequest {
     int32 cltv_limit = 9;
 }
 
-message PaymentResponse {
+message SendPaymentResponse {
+}
+
+message LookupPaymentRequest {
+    /// The hash of the payment to look up.
+    bytes payment_hash = 1;
+}
+
+enum PaymentState {
```

Comment:
What about "terminal" failures like unknown payment hash?

Reply:
The plan is to add that in a follow up pr. `Router` isn't yet storing those outcomes. The follow up pr would add storing them and passing them through over rpc.